lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -91,60 +90,68 @@ export default function BasicLtiLaunchApp({ rpcServer }) {
    *  "error-authorizing", or "error-report-submission"
    * @param {boolean} [retry=true] - Can the request be retried?
    */
-  const handleError = (e, newErrorState, retry = true) => {
+  const handleError = useCallback((e, newErrorState, retry = true) => {
     if (e instanceof ApiError && !e.errorMessage && retry) {
       setErrorState('error-authorizing');
     } else {
       setError(e);
       setErrorState(newErrorState);
     }
-  };
+  }, []);
 
   /**
    * Fetch the groups from the sync endpoint if `sync` object exists.
+   *
+   * @return {boolean} - true if an error occurred, false otherwise
```

Comment:
Logically it seems confusing to me to have a function called `fetchGroups` which returns `true` if it _failed_ rather than succeeded. The same applies to `fetchContentUrl` below.